pkgs/tools/system/thermald/default.nix

```diff
@@ -1,26 +1,48 @@
-{ stdenv, fetchFromGitHub, autoconf, automake, libtool
-, pkgconfig, dbus, dbus-glib, libxml2, autoconf-archive }:
+{ autoconf
+, autoconf-archive
+, automake
+, dbus
+, dbus-glib
+, fetchFromGitHub
+, gtk-doc
+, libevdev
+, libtool
+, libxml2
+, lzma
+, pkgconfig
+, stdenv
+, upower
+}:
 
 stdenv.mkDerivation rec {
   pname = "thermald";
-  version = "2.2";
+  version = "2.3";
 
   src = fetchFromGitHub {
     owner = "intel";
     repo = "thermal_daemon";
     rev = "v${version}";
-    sha256 = "1nrhv3bypyc48h9smj5cpq63rawm6vqyg3cwkhpz69rgjnf1283m";
+    sha256 = "0cisaca2c2z1x9xvxc4lr6nl6yqx5bww6brh73m0p1n643jgq1dl";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ autoconf automake libtool dbus dbus-glib libxml2 autoconf-archive ];
-
-  patchPhase = ''sed -e 's/upstartconfdir = \/etc\/init/upstartconfdir = $(out)\/etc\/init/' -i data/Makefile.am'';
+  nativeBuildInputs = [
+    gtk-doc
+    pkgconfig
+  ];
+  buildInputs = [
+    autoconf
+    autoconf-archive
+    automake
+    dbus
+    dbus-glib
+    libevdev
+    libtool
+    libxml2
+    lzma
+    upower
+  ];
 
-  preConfigure = ''
-    export PKG_CONFIG_PATH="${dbus.dev}/lib/pkgconfig:$PKG_CONFIG_PATH"
-    ./autogen.sh
-  '';
+  preConfigure = "./autogen.sh";
```

Comment:
This will make configure run twice:

https://github.com/intel/thermal_daemon/blob/a86a9fc1bbc3d44c452adf69a0e2a8df8933e165/autogen.sh#L15-L17

Reply:
should be fixed by using `preConfigure = "NO_CONFIGURE=1 ./autogen.sh"; `

Reply:
Isn't that what we have `autoreconfHook` for?

Reply:
[`autoreconfHook`](https://github.com/NixOS/nixpkgs/blob/aaf1331bca72b60750bb69624a37f2aa81063e05/pkgs/build-support/setup-hooks/autoreconf.sh) just runs `autoreconf -vfi`, which might not be sufficient in some projects. (Does it run `gtkdocize` like `autogen.sh` does? Is that necessary?)